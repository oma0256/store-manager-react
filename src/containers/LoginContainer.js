import React, { Component } from 'react';
import { connect } from 'react-redux';
import Login from '../components/Login';
import { loginFetch, dismissLoginError } from '../store/actions/login';

export class LoginContainer extends Component {
  state = {
    email: '',
    password: '',
    disabled: true
  };

  UNSAFE_componentWillUpdate = (nextProps, nextState) => {
    const { email, password } = nextState;
    if (email !== this.state.email || password !== this.state.password) {
      const disabled = email && password ? false : true;
      this.setState({ disabled });
    }
  };

  onChangeHandler = e => this.setState({ [e.target.name]: e.target.value });

  onSubmitHandler = e => {
    e.preventDefault();
    const { email, password } = this.state;
    const payload = {
      data: {
        email,
        password
      },
      history: this.props.history
    };
    this.props.loginFetch(payload);
  };

  onDismissHandler = () => this.props.dismissLoginError();

  render() {
    const loginProps = {
      ...this.state,
      onChangeHandler: this.onChangeHandler,
      onSubmitHandler: this.onSubmitHandler,
      onDismissHandler: this.onDismissHandler
    };

    return <Login {...loginProps} />;
  }
}

const mapDispatchToProps = dispatch => {
  return {
    loginFetch: payload => dispatch(loginFetch(payload)),
    dismissLoginError: () => dispatch(dismissLoginError())
  };
};

export default connect(
  null,
  mapDispatchToProps
)(LoginContainer);
