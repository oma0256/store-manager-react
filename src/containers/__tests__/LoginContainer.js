import React from 'react';
import { shallow } from 'enzyme';
import { LoginContainer } from '../LoginContainer';

describe('Login Container', () => {
  const loginFetchMock = jest.fn();
  const dismissLoginErrorMock = jest.fn();
  const loginContainerProps = {
    loginFetch: loginFetchMock,
    dismissLoginError: dismissLoginErrorMock
  };
  const loginContainer = shallow(<LoginContainer {...loginContainerProps} />);
  const loginContainerInstance = loginContainer.instance();

  it('renders correctly', () => {
    expect(loginContainer).toMatchSnapshot();
  });

  it('onSubmitHandler dispatches loginFetch action creator', () => {
    loginContainerInstance.onSubmitHandler({ preventDefault: jest.fn() });
    expect(loginFetchMock).toHaveBeenCalled();
  });

  it('onDismissHandler dispatches dismissLoginError action creator', () => {
    loginContainerInstance.onDismissHandler();
    expect(dismissLoginErrorMock).toHaveBeenCalled();
  });

  it('onChangeHandler changes state when called', () => {
    loginContainerInstance.onChangeHandler({
      target: { name: 'email', value: 'okay@email.com' }
    });
    expect(loginContainer.state().email).toBe('okay@email.com');
    loginContainerInstance.onChangeHandler({
      target: { name: 'password', value: 'okay' }
    });
    expect(loginContainer.state().password).toBe('okay');
  });
});
