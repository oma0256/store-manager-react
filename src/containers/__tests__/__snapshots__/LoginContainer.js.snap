// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container renders correctly 1`] = `ShallowWrapper {}`;
